### ISSUE TITLE
On agent-auth step "ERROR: Unable to connect to <My_Public_IP>:1515"

### ISSUE THREAD

**Issue by vaibhavpandey86** [wazuh/wazuh#6079]
Hello,

I do have wazuh manager running in GCP, installed and configured successfully. one client as well in the same GCP working fine. GUI working fine, accessible through my laptop.( out of GCP)
wazuh manager and client version : 3.13.1
When I am trying to make my laptop( out of GCP ) a client I am getting this error.

sh-3.2# ./agent-auth -m <My_Public_IP>
2020/09/23 10:12:59 agent-auth: INFO: Started (pid: 3284).
2020/09/23 10:12:59 agent-auth: INFO: No authentication password provided.
2020/09/23 10:13:43 agent-auth: ERROR: Unable to connect to <My_Public_IP>:1515

this manager is reachable through ping : 
sh-3.2# ping <My_Public_IP>
PING <My_Public_IP> (<My_Public_IP>): 56 data bytes
64 bytes from <My_Public_IP>: icmp_seq=0 ttl=50 time=323.176 ms

ossec-authd process is running:
root@wazuhmanager:/home/vaibhav# ps -aux | grep ossec-authd
root      113794  0.2  0.1 166680  4812 ?        Sl   Sep22   2:21 /var/ossec/bin/ossec-authd

Please help me out!!!

Thanks,

**Comment by MarcelKemp** [comment]
Hi vaibhav,

It looks like a networking problem.
I suggest you review the security policies of the gcp environment and check if the ports are blocked by the firewall. Normally, the manager uses port 1515 to try to authenticate the agent.

Also, I recommend that you try to register the agent manually, with the manage_agents tool. You can follow the steps in the [documentation](https://documentation.wazuh.com/3.13/user-manual/registering/command-line-registration.html#command-line-registration).

Have you seen an error in the manager's log? If yes, please send it to me so I can take a look and help you better.

If the error persists, do not hesitate to contact us.

**Comment by vaibhavpandey86** [comment]
Ok, let me open the 1515 port.
Could you please tell me the path for manager's log?

**Comment by MarcelKemp** [comment]
Hello again,

Once you have successfully registered the agents, you can close it again by disabling the authentication service in the manager, as you can see in the [documentation](https://documentation.wazuh.com/3.13/user-manual/reference/ossec-conf/auth.html#disabled). But keep in mind that if you want to re-register new agents in the future, you will have to re-enable it.

Also, in the manager, you can find all the logs in the following path: `/var/ossec/logs/` . In that folder, you can view all the logs and alerts generated. The most used and containing more information is the log: `ossec.log` .

Was the agent able to connect successfully?
Do you have any further questions?

**Comment by vaibhavpandey86** [comment]
Hello... so I am trying with different combination like... whether I need to open TCP or UDP ports.... 1514/1515
As in the other agent "communication" conf. it says manager available <Public_IP> <UDP Port 1514>.
so is it like we need 1515 to connection establishment and 1514 for communication?
till now I have opened these ports :
tcp:1514 
tcp:1515 
udp:1514
so new agent is listed in the agent list but not active, still after multiple restart.
I will try with all four ports open.

**Comment by MarcelKemp** [comment]
So, you have been able to register the agent in the manager, right?

And yes, it is necessary to have ports 1515 open to be able to register the agent, and 1514 to establish communication between the manager and the agent.

If the problem between the connection of the agent and the manager persists, could you send me the logs of the manager and the agent to try to help you better?

**Comment by vaibhavpandey86** [comment]
Hello Marcel, can I conclude that I need to keep TCP(not UDP) port 1514 and 1515 open all the time. As I tried closing the ports, my agent got disconnected.

**Comment by vaibhavpandey86** [comment]
I tried to add new agent. Earlier it  was added and working fine ( through agent-auth ), then I removed it and tried to add with key:
* Wazuh v3.13.1 Agent manager.         *
* The following options are available: *
****************************************
   (A)dd an agent (A).
   (E)xtract key for an agent (E).
   (L)ist already added agents (L).
   (R)emove an agent (R).
   (Q)uit.
Choose your action: A,E,L,R or Q: E

Then restarted the agent ./ossec-control restart

sh-3.2# ./ossec-control restart
wazuh-modulesd not running...
ossec-logcollector not running...
ossec-syscheckd not running...
ossec-agentd not running...
ossec-execd not running...
Wazuh v3.13.2 Stopped
Starting Wazuh v3.13.2...
ossec-execd did not start

It is not connecting.
Ossec.log from manager :
root@wazuhmanager:/var/ossec/bin# tail -f ../logs/ossec.log 
2020/09/28 11:41:29 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:42:34 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:43:04 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 11:43:04 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2020/09/28 11:43:39 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:44:44 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:45:49 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:46:54 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').
2020/09/28 11:47:54 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Red Hat Enterprise Linux' database update.
2020/09/28 11:47:59 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '49.207.204.30' (name 'unknown').


Ossec.log for Mac agent :
sh-3.2# tail -f ../logs/ossec.log 
2020/09/28 17:05:30 agent-auth: INFO: Using agent name as: exium.local
2020/09/28 17:05:30 agent-auth: INFO: Request sent to manager
2020/09/28 17:05:30 agent-auth: INFO: Waiting for manager reply
2020/09/28 17:05:30 agent-auth: INFO: Received response with agent key
2020/09/28 17:05:30 agent-auth: INFO: Valid key created. Finished.
2020/09/28 17:05:30 agent-auth: INFO: Connection closed.
2020/09/28 17:05:56 ossec-execd: INFO: Started (pid: 69395).
2020/09/28 17:05:56 ossec-execd: CRITICAL: (1210): Queue '/Library/Ossec/queue/alerts/execq' not accessible: 'No such file or directory'
2020/09/28 17:17:18 ossec-execd: INFO: Started (pid: 71388).
2020/09/28 17:17:18 ossec-execd: CRITICAL: (1210): Queue '/Library/Ossec/queue/alerts/execq' not accessible: 'No such file or directory'

**Comment by vaibhavpandey86** [comment]
**Also seems like my vulnerability scanning module and email alerts is not working:**
2020/09/28 12:12:36 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2020/09/28 12:12:53 wazuh-modulesd:vulnerability-detector: WARNING: (5522): There was no valid response to 'https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-2010.meta' after '3' attempts.
2020/09/28 12:12:53 wazuh-modulesd:vulnerability-detector: WARNING: (5500): The 'National Vulnerability Database' database could not be fetched.
2020/09/28 12:12:53 wazuh-modulesd:vulnerability-detector: ERROR: (5513): CVE database could not be updated.
2020/09/28 12:16:33 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 12:16:33 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2020/09/28 12:17:54 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2020/09/28 12:18:07 ossec-maild: ERROR: (1223): Error Sending email to 108.177.111.27 (smtp server)
2020/09/28 12:18:10 wazuh-modulesd:vulnerability-detector: WARNING: (5522): There was no valid response to 'https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-2010.meta' after '3' attempts.
2020/09/28 12:18:10 wazuh-modulesd:vulnerability-detector: WARNING: (5500): The 'National Vulnerability Database' database could not be fetched.
2020/09/28 12:18:10 wazuh-modulesd:vulnerability-detector: ERROR: (5513): CVE database could not be updated.
2020/09/28 12:21:34 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 12:21:34 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2020/09/28 12:26:35 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 12:26:35 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2020/09/28 12:31:36 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 12:31:36 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2020/09/28 12:36:37 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2020/09/28 12:36:37 wazuh-modulesd:vulnerability-detector: ERROR: (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.

**Comment by MarcelKemp** [comment]
Sorry for the delay,

First of all, port 1515 does not need to be open once the agent has registered, however port 1514 does, because that is where the manager and agent communicate.

Also for port 1514, it will depend on how you have configured the connection between both machines in ossec.conf [<client>], since by default it is set to UDP, so it would be necessary to keep port 1514/UDP open. [Agent connection](https://documentation.wazuh.com/3.13/user-manual/agents/agent-connection.html)

According to the manual registration of the agent, what I can see is that you have extracted the key, so the next necessary step would be to import the key into the agent, as can be seen below:

****************************************
* Wazuh v4.0.0 Agent manager.          *
* The following options are available: *
****************************************
   (I)mport key from the server (I).
   (Q)uit.
Choose your action: I or Q: I
* Provide the Key generated by the server.
* The best approach is to cut and paste it.
*** OBS: Do not include spaces or new lines.
Paste it here (or '\q' to quit): Insert here the key extracted from the manager's manage_agents

After registering, you need to restart the agent (systemctl restart wazuh-agent).

And the VDT error, the only thing necessary to fix the error is to update wazuh to version 1.13.2, as explained by my workmate in the following [issue](https://github.com/wazuh/wazuh/issues/6118#issuecomment-700485714).

**Comment by vaibhavpandey86** [comment]
Hello @MarcelKemp 

Thank you so much for your great help. I tried the same steps yesterday, It was not working... but it worked magically today. so agent problem solved.

For NVD issue I am following the same steps mentioned above, but still no luck, so i have created separate issue for the same and keeping an eye on it [https://github.com/wazuh/wazuh/issues/6118](url)
I am closing this one for now.
Thanks much again!!! 👍🏼 
Vaibhav

**Comment by vaibhavpandey86** [comment]
Hello @MarcelKemp ,

any Help with this error :

 ./ossec-control restart
wazuh-modulesd not running...
ossec-logcollector not running...
ossec-syscheckd not running...
ossec-agentd not running...
ossec-execd not running...
Wazuh v3.13.2 Stopped
Starting Wazuh v3.13.2...
**ossec-execd did not start**

**Ossec directory permission :**
drwxr-x---  14 root  ossec             448 Sep 25 18:31 Ossec

**ossec.log**

2020/09/30 12:17:25 agent-auth: INFO: Started (pid: 15456).
2020/09/30 12:17:25 agent-auth: INFO: Starting enrollment process to server: <public_IP>
2020/09/30 12:17:25 agent-auth: INFO: Connected to <public_IP>:1515
2020/09/30 12:17:25 agent-auth: INFO: Registering agent to unverified manager.
2020/09/30 12:17:25 agent-auth: INFO: No authentication password provided.
2020/09/30 12:17:25 agent-auth: INFO: Using agent name as: exium.local
2020/09/30 12:17:25 agent-auth: INFO: Request sent to manager
2020/09/30 12:17:25 agent-auth: INFO: Waiting for manager reply
2020/09/30 12:17:26 agent-auth: INFO: Received response with agent key
2020/09/30 12:17:26 agent-auth: INFO: Valid key created. Finished.
2020/09/30 12:17:26 agent-auth: INFO: Connection closed.
2020/09/30 13:03:22 ossec-execd: INFO: Started (pid: 19930).
2020/09/30 13:03:22 ossec-execd: CRITICAL: (1210): Queue '/Library/Ossec/queue/alerts/execq' not accessible: 'No such file or directory'
2020/09/30 13:06:51 ossec-execd: INFO: Started (pid: 20615).
2020/09/30 13:06:51 ossec-execd: CRITICAL: (1210): Queue '/Library/Ossec/queue/alerts/execq' not accessible: 'No such file or directory'

**Comment by vaibhavpandey86** [comment]
Hello @MarcelKemp  sorry for inconvenience . my problem solved. thank you 👍🏼

**Comment by MarcelKemp** [comment]
Hello @vaibhavpandey86,

I am happy that you have finally solved the problem.

We would appreciate it if you would open new topics to discuss problems of different origin than the original, just to keep the repository clean.

I also recommend that you send your questions and concerns to the [Wazuh mailing lists](https://groups.google.com/g/wazuh) or the [Slack #community channel](https://wazuh.slack.com/), where we can help you more dynamically, that way we can let the repository to report product bugs or feature requests.

Hope you have a good day.

**Comment by biancaasan** [comment]
Hello. I have installed a wazuh agent on Ubuntu and my wazuh server on another. When I try to connect my agent to the server I receive the folowing error:
2021/02/26 11:39:17 agent-auth: INFO: Started (pid: 2855).
2021/02/26 11:39:18 agent-auth: INFO: No authentication password provided.
2021/02/26 11:41:28 agent-auth: ERROR: Unable to connect to IP:1515

Can you help me, please?

**Comment by vaibhavpandey86** [comment]
> Hello. I have installed a wazuh agent on Ubuntu and my wazuh server on another. When I try to connect my agent to the server I receive the folowing error:
> 2021/02/26 11:39:17 agent-auth: INFO: Started (pid: 2855).
> 2021/02/26 11:39:18 agent-auth: INFO: No authentication password provided.
> 2021/02/26 11:41:28 agent-auth: ERROR: Unable to connect to IP:1515
> 
> Can you help me, please?

Hello @biancaasan .. I followed the above steps mentioned. and I think I solved my issue. you need to open the ports... if you are in cloud environment or any physical computer.
It's a long back but I followed them only.

Thanks,
Vaibhav

**Comment by biancaasan** [comment]
@vaibhavpandey86 I opened the ports and it worked. Thank you!

**Comment by hansimehdi** [comment]
Make sure that your agent hostname and manager hostname are not the same.
